host/platform/Windows/src/platform.c

```diff
@@ -397,3 +399,24 @@ HWND os_getMessageWnd(void)
 {
   return app.messageWnd;
 }
+
+bool os_blockScreensaver()
+{
+  static bool      lastResult = false;
+  static ULONGLONG lastCheck  = 0;
+
+  ULONGLONG now = GetTickCount64();
```

Comment:
Do we really need to poll this? Isn't there a windows power API we can register events for?

Reply:
Yes, there are power APIs that would happily give events for when things happen, like system suspend (`PowerRegisterSuspendResumeNotification`) and monitors turns on and off, power status, and whether the user activity timed out ([GUIDs](https://docs.microsoft.com/en-us/windows/win32/power/power-setting-guids) for PowerSettingRegisterNotification).

But I looked and looked, and I can't find *any* API other than `CallNtPowerInformation` that would report whether `SetThreadExecutionState` is used. I would gladly use an event-based API, but I can't find one.

Reply:
Not a problem then, just making sure the option was explored first ;)